generated/nifake/unit_tests/test_session.py

```diff
@@ -680,6 +680,16 @@ def test_device_method_not_exist_on_repeated_capability_error(self):
             except AttributeError:
                 pass
 
+    def test_repeated_capabilities_list(self):
+        with nifake.Session('dev1') as session:
+            assert session.channels['0']._repeated_capability_list == ['0']
+            assert session.channels['r0']._repeated_capability_list == ['r0']
+            assert session.channels['0,1']._repeated_capability_list == ['0', '1']
+            assert session.channels['0-2']._repeated_capability_list == ['0', '1', '2']
+            assert session.channels['2-0']._repeated_capability_list == ['2', '1', '0']
```

Comment:
Missing: test for colon as well.

I will say, however, that a lot of this is redundant testing because the conversion function is already tested. You could simply mock the conversion function and make sure that:

* The parameters make it correctly to the function
* They are used correctly by our client code
* Errors are handled properly by our client code

You end up with 3 tests.

Reply:
Same as above. We have one test here now.